mkosi/__init__.py

```diff
@@ -1331,6 +1376,17 @@ def mount_image(
         yield
     finally:
         with complete_step("Unmounting image"):
+            if realroot is not None:
+                umount(root)
+                umount(realroot)
+
+                workdir.cleanup()
+
+                # Let's now move the contents of realroot into root
+                for entry in os.scandir(realroot):
+                    os.rename(realroot / entry.name, root / entry.name)
+                realroot.rmdir()
```

Comment:
i am pretty sure this rename() loop should be under a new `with complete_step()` since it's slow and that deserves a spearate explanation given this is not really "unmounting" anymore

Reply:
maybe move the body of this into a new helper function: move_dir_contents() or so?